x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfig.java

```diff
@@ -60,6 +64,45 @@
     private static final int TWO_MINS_SECONDS = 2 * SECONDS_IN_MINUTE;
     private static final int TWENTY_MINS_SECONDS = 20 * SECONDS_IN_MINUTE;
     private static final int HALF_DAY_SECONDS = 12 * 60 * SECONDS_IN_MINUTE;
+    static final XContentObjectTransformer<QueryBuilder> QUERY_TRANSFORMER = XContentObjectTransformer.queryBuilderTransformer();
+    private static final BiFunction<Map<String, Object>, String, QueryBuilder> lazyQueryParser = (objectMap, id) -> {
+        try {
+            return QUERY_TRANSFORMER.fromMap(objectMap);
+        } catch (IOException | XContentParseException exception) {
+            // Certain thrown exceptions wrap up the real Illegal argument making it hard to determine cause for the user
+            if (exception.getCause() instanceof IllegalArgumentException) {
+                throw ExceptionsHelper.unprocessableEntityException(
```

Comment:
This and other function handlers throw `ExceptionsHelper.unprocessableEntityException`. Should this be kept as such or move to `XContentParseException` or possibly `IllegalArgumentException` ? So as not to dictate the RESTful response code at this layer.

Reply:
I think that for BWC the aim should be to keep the exceptions for invalid syntax from a PUT datafeed the same as they were in 6.5.  So `unprocessableEntityException` (422) is almost certainly not right.

Please can you do an experiment and check what exception gets thrown if you do a PUT datafeed with an invalid query or aggregation in the datafeed using the master branch code.  Then change this code so the end user experience is identical when calling PUT datafeed.